### PR TITLE
docs: add documentation for the RabbitMQ bridge

### PR DESCRIPTION
**Pull request by kjellwinblad**

### CODE REVIEW

**Comment by LenaLenaPan** [comment]
@kjellwinblad can you also add a milestone to indicate the product version it will be released?

**Comment by kjellwinblad** [comment]
@LenaLenaPan I have set the milstone e5.0.3 as I think that is the next Enterprise release and I believe the RabbitMQ bridge will be included there. @id can you change this if e5.0.3 is not the correct version?

**Comment by LenaLenaPan** [comment]
@kjellwinblad just checked with Shifan and guowei, it will be released with 5.0.4, will update the milestone to 5.0.4

File: en_US/data-integration/data-bridge-rabbitmq.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+After setting up the RabbitMQ server using the Docker image, you can create a test exchange and a queue using the RabbitMQ management web interface. Here we will describe how to set this up using the RabbitMQ RabbitMQ Management Web Interface. You can skip this if you already have an exchange and queue to test with.
+
+1. **Access the RabbitMQ Management Web Interface**: Open your web browser and navigate to http://localhost:15672/. You should see the RabbitMQ management web interface login page. Enter the default credentials: Username: guest and Password: guest, then click on "Log in".
+2. **Create a Test Exchange**: After logging in, click on the "Exchanges" tab in the top menu. In the "Add a new exchange" section at the bottom of the page, input the following information:
+   * **Name**: test_exchange
+   * **Type**: Choose direct from the drop-down list
+   * **Durability**: Choose Durable to make the exchange persistent
+   * **Auto-delete**: Leave unchecked
+   * **Internal**: Leave unchecked
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-rabbitmq.md]:
```suggestion
   * **Internal**: No
```

File: en_US/data-integration/data-bridge-rabbitmq.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+### Create a RabbitMQ Test Exchange and Queue
+
+After setting up the RabbitMQ server using the Docker image, you can create a test exchange and a queue using the RabbitMQ management web interface. Here we will describe how to set this up using the RabbitMQ RabbitMQ Management Web Interface. You can skip this if you already have an exchange and queue to test with.
+
+1. **Access the RabbitMQ Management Web Interface**: Open your web browser and navigate to http://localhost:15672/. You should see the RabbitMQ management web interface login page. Enter the default credentials: Username: guest and Password: guest, then click on "Log in".
+2. **Create a Test Exchange**: After logging in, click on the "Exchanges" tab in the top menu. In the "Add a new exchange" section at the bottom of the page, input the following information:
+   * **Name**: test_exchange
+   * **Type**: Choose direct from the drop-down list
+   * **Durability**: Choose Durable to make the exchange persistent
+   * **Auto-delete**: Leave unchecked
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-rabbitmq.md]:
```suggestion
   * **Auto-delete**: No
```

File: en_US/data-integration/data-bridge-rabbitmq.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+   * **Internal**: Leave unchecked
+   * **Alternate-exchange**: Leave empty
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-rabbitmq.md]:
```suggestion
```

File: en_US/data-integration/data-bridge-rabbitmq.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+   * **Name**: test_queue
+   * **Durability**: Choose Durable to make the queue persistent
+   * **Auto-delete**: Leave unchecked
+   * **Message TTL**: Leave empty
+   * **Expires**: Leave empty
+   * **Max length**: Leave empty
+   * **Max length** bytes: Leave empty
+   * **Max priority**: Leave empty
+   * **Dead letter exchange**: Leave empty
+   * **Dead letter routing key**: Leave empty
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-rabbitmq.md]:
```suggestion
```

File: en_US/data-integration/data-bridge-rabbitmq.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+5. Click on the "Add queue" button to create the test queue.
+6. **Bind the Test Queue to the Test Exchange**: After creating the test queue, click on its name in the "Queues" tab to open its details page. Click on the "Bindings" tab under the queue name. In the "Add binding from an exchange" section, input the following information:
+   * **From**: Choose test_exchange from the drop-down list
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-rabbitmq.md]:
```suggestion
   * **From exchange**: test_exchange
```

File: en_US/data-integration/data-bridge-rabbitmq.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+   * In the **Delivery Mode** dropdown, choose between `non_persistent` and `persistent`:
+       * `non_persistent` (default): Messages are not persisted to disk and may be lost if RabbitMQ restarts or crashes.
+       * `persistent`: Messages are persisted to disk, providing durability in case RabbitMQ restarts or crashes. Notice that you may also need to set the queue and exchange as durable to prevent messages from being lost in case RabbitMQ is restarted. See the documentation of RabbitMQ for more information.
+   * **Enable Publish Confirmations**: To ensure that messages are successfully published to RabbitMQ, you can enable publish confirmations. This feature ensures that the RabbitMQ broker acknowledges the receipt of a published message before considering it successfully published. Enabling publish confirmations can help improve the reliability of your message delivery. To enable publish confirmations, set the wait_for_publish_confirmations configuration option to true. By default, this option is set to true, so publish confirmations are enabled by default.
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-rabbitmq.md]:
```suggestion
   * **Wait for Publish Confirmations**: To ensure that messages are successfully published to RabbitMQ, you can enable publish confirmations. This feature ensures that the RabbitMQ broker acknowledges the receipt of a published message before considering it successfully published. Enabling publish confirmations can help improve the reliability of your message delivery. To enable publish confirmations, set the "Wait for Publish Confirmations" configuration option to true. By default, this option is set to true, so publish confirmations are enabled by default.
```

File: en_US/data-integration/data-bridge-rabbitmq.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+   * **Set Publish Confirmation Timeout**: The publish confirmation timeout determines the duration the publisher will wait for the broker's acknowledgment before considering the publish operation a failure.
+   * **Payload Template**: The payload template field allows you to define a custom message payload format that will be sent to the RabbitMQ exchange. You can use placeholders within the template to dynamically include data from the incoming MQTT messages. These placeholders are enclosed in `${}` and will be replaced by the actual values from the message when it is forwarded to the RabbitMQ server. For example, if you want to include the MQTT message payload and its timestamp in the RabbitMQ message, you can use the template like this: {"payload": "${data}", "timestamp": ${timestamp}}. This template will produce a JSON-formatted message containing the payload and timestamp of the incoming MQTT message. The default value for the payload template field is an empty string, which means the message payload will be forwarded to RabbitMQ without any modification as JSON formatted text.
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-rabbitmq.md]:
```suggestion
   * **Payload Template**: The payload template field allows you to define a custom message payload format that will be sent to the RabbitMQ exchange. You can use placeholders within the template to dynamically include data from the incoming MQTT messages. These placeholders are enclosed in `${}` and will be replaced by the actual values from the message when it is forwarded to the RabbitMQ server. For example, if you want to include the MQTT message payload and its timestamp in the RabbitMQ message, you can use the template like this: {"payload": "${payload}", "timestamp": ${timestamp}}. This template will produce a JSON-formatted message containing the payload and timestamp of the incoming MQTT message. The default value for the payload template field is an empty string, which means the message payload will be forwarded to RabbitMQ without any modification.
```

File: en_US/data-integration/data-bridge-rabbitmq.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+1. Fill in the connection information for the current EMQX instance. If you are running EMQX locally, you can use the default values unless you have changed EMQX's default configuration (for example, you might have configured authentication which may require you to type in a username and password).
+2. Click **Connect** to connect the client to the EMQX instance.
+3. Scroll down to the publish area and type in the following:
+   * **Topic**: t/test
+   * **Payload**: Hello World RabbitMQ from EMQX
+   * **QoS**: 2
+4. Click **Publish** to send the message.
+
+If everything has gone according to plan, a message should have been published to the specified exchange in the RabbitMQ server with the specified routing key. You can check this by visiting the RabbitMQ Management Console at http://localhost:15672 (use guest as both username and password if you haven't changed the default settings) and navigating to the Queues section to verify the message has been routed to the appropriate queue(s).
```
Comment (kjellwinblad) [inline en_US/data-integration/data-bridge-rabbitmq.md]:
```suggestion
If everything has gone according to plan, a message should have been published to the specified exchange in the RabbitMQ server with the specified routing key. You can check this by visiting the RabbitMQ Management Console at http://localhost:15672 (use guest as both username and password if you haven't changed the default settings) and navigating to the Queues section to verify the message has been routed to the appropriate queue(s). To see the message content, click on the the queue to see details and then on the "Get Message(s)" button.
```

**Review by wivwiv — APPROVED** [review]

**Pull request merged**